azure-quantum/azure/quantum/_authentication/_default.py

```diff
@@ -84,22 +87,86 @@ def _authority_or_default(self, authority: str, arm_endpoint: str):
             return ConnectionConstants.DOGFOOD_AUTHORITY
         return ConnectionConstants.AUTHORITY
 
-    def _initialize_credentials(self):
+    def _get_cache_options(self) -> Optional[TokenCachePersistenceOptions]:
+        """
+        Returns a valid TokenCachePersistenceOptions
+        if the AzureIdentity Persistent Cache is accessible.
+        Returns None otherwise.
+        """
+        cache_options = TokenCachePersistenceOptions(
+            allow_unencrypted_storage=False,
+            name="AzureQuantumSDK"
+        )
+        try:
+            # pylint: disable=protected-access
+            cache = AzureIdentityPersistentCache._load_persistent_cache(cache_options)
+            try:
+                # Try to get the location of the cache for
+                # tracing purpose.
+                _LOGGER.info(
+                    "Using Azure.Identity Token Cache at %s.",
+                    cache._persistence.get_location()
+                )
+            except: # pylint: disable=bare-except
+                _LOGGER.info("Using Azure.Identity Token Cache.")
+            return cache_options
+        except Exception as ex: # pylint: disable=broad-except
+            if (isinstance(ex, ValueError)
```

Comment:
@kikomiss, hopefully this can provide additional guidance for users facing issues with `libsecret`, but it's a best effort try, and I like your idea of having some documentation in Microsoft Learn docs.

Reply:
If we want to check if `libsecret` is working (instead of checking a `ValueError` message that can change in the future) we could call [msal_extensions.trial_run()](https://github.com/AzureAD/microsoft-authentication-extensions-for-python/blob/ae07f283bb6e7d601f5605a87be4d32b36828593/msal_extensions/libsecret.py#L110) if `sys.platform.startswith("linux")`.

Reply:
Again, I implemented my own comment :-)